### DISCUSSION TITLE
Permission to run script not working

### DISCUSSION THREAD

**Discussion by gabrielat** [netbox-community/netbox#14377] [Help Wanted!] [answered]
Hi! I've setup this permissions so the user can do everything + run in specific features, and scripts are one of them. However, when I try to run scripts with that user, I get a message that I don't have any scripts and to try to add custom scripts. And when I try to add a new script, I get hte message that I don't have permission to do that.
Here are some prints..
![image](https://github.com/netbox-community/netbox/assets/14821425/12636e35-1256-45bd-b7e4-aa9eee045e0b)
![image](https://github.com/netbox-community/netbox/assets/14821425/88750fa1-6a2a-408f-b195-26fb2890d802)
![image](https://github.com/netbox-community/netbox/assets/14821425/a86a557d-3659-43e4-930f-c689b2209384)

Thanks in advance!

**Comment by candlerb** [comment]
It looks like you added this permission to "Extras > Saved Filter" and "Extras > Tag", but not to "Extras > Script"

The "run" permission doesn't make any sense with those other types, although it doesn't do any harm to have it.  But if it were me, I'd create a brand new permission for "Extras > Script", set it up how you need (including "run" permission), and then add that permission to each group and/or user who needs it.

**Reply by gabrielat** [reply]
Hi, thank you for your reply
Actually, I have selected these permissions:
![image](https://github.com/netbox-community/netbox/assets/14821425/db5caf45-ab5e-4b24-b6fd-8f99df4165bc)

**Reply by candlerb** [reply, marked as answer]
According to the [documentation](https://docs.netbox.dev/en/stable/customization/custom-scripts/#running-custom-scripts):

> To run a custom script, a user must be assigned via permissions for Extras > Script, Extras > ScriptModule, and Core > ManagedFile objects. They must also be assigned the extras.run_script permission. This is achieved by assigning the user (or group) a permission on the Script object and specifying the run action

The screenshot you showed initially appeared to be a Permission which was assigned to object types "Extras > Saved Filter" and "Extras > Tag" only.

The screenshot you have just shown I'm guessing is from a list of Permissions, with a permission called "NUC" which is bound to object type "Script" and "Script Module", but not "Managed File" AFAICS.

If I create such a permission, it looks like this in the UI (Netbox v3.6.5):

![image](https://github.com/netbox-community/netbox/assets/44789/df06f742-aef2-4d63-98df-362e401c1d67)

EDIT: the documentation doesn't say *which* permissions need to be assigned to those objects. I guess just "View" would be sufficient. I see no reason for an end-user who needs only to run a script, to be able to add, change or delete scripts.

**Reply by gabrielat** [reply]
Thank you! I was missing the Core > ManagedFiles permission!
